presto-main/src/main/java/com/facebook/presto/operator/aggregation/ArrayAggregation.java

```diff
@@ -0,0 +1,202 @@
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(ArrayAggregation.class.getClassLoader());
+
+        ArrayAggregationStateSerializer stateSerializer = new ArrayAggregationStateSerializer();
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        List<Type> inputTypes = ImmutableList.of(valueType);
+        Type outputType = new ArrayType(valueType);
+        ArrayAggregationStateFactory stateFactory = new ArrayAggregationStateFactory(valueType);
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, valueType, inputTypes),
+                createInputParameterMetadata(valueType),
+                INPUT_FUNCTION,
+                null,
+                null,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                ArrayAggregationState.class,
+                stateSerializer,
+                stateFactory,
+                outputType,
+                false);
+
+        GenericAccumulatorFactoryBinder factory = new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, intermediateType, valueType, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type value)
+    {
+        return ImmutableList.of(new ParameterMetadata(STATE), new ParameterMetadata(INPUT_CHANNEL, value), new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    public static void input(ArrayAggregationState state, Block value, int position)
+    {
+        if (state.getSliceOutput() == null) {
+            SliceOutput sliceOutput = new DynamicSliceOutput(400);
+            state.setEntries(0);
+            state.setSliceOutput(sliceOutput);
+        }
+        state.setEntries(state.getEntries() + 1);
+        appendTo(state.getType(), state.getSliceOutput(), value, position);
+    }
+
+    private static void appendTo(Type type, SliceOutput output, Block block, int position)
+    {
+        if (type.getJavaType() == long.class) {
+            output.appendLong(type.getLong(block, position));
+        }
+        else if (type.getJavaType() == double.class) {
+            output.appendDouble(type.getDouble(block, position));
+        }
+        else if (type.getJavaType() == Slice.class) {
+            Slice s = type.getSlice(block, position);
+            output.appendInt(s.length());
+            output.appendBytes(s);
+        }
+        else if (type.getJavaType() == boolean.class) {
+            output.appendByte(type.getBoolean(block, position) ? 1 : 0);
+        }
+        else {
+            throw new IllegalArgumentException("Unsupported type: " + type.getJavaType().getSimpleName());
+        }
+    }
+
+    public static void combine(ArrayAggregationState state, ArrayAggregationState otherState)
+    {
+        SliceOutput s1 = state.getSliceOutput();
```

Comment:
I think we try to avoid abbreviations in general, so name this something like 'slice' and 'otherSlice' to match the parameters